gulpfile.js

```diff
@@ -1,115 +1,142 @@
 'use strict';
 
 var gulp = require('gulp'),
-  fs = require('fs-extra'),
-  q = require('q'),
-  request = require('request'),
-  spawn = require('child_process').spawn,
-  $ = require('gulp-load-plugins')();
-
-var cleanDirectories = function(directories) {
-  directories.forEach(function(directory) {
-    fs.removeSync(directory);
-    fs.mkdirsSync(directory);
-  });
+	runSequence = require('run-sequence'),
+    fs = require('fs-extra'),
+    q = require('q'),
+    request = require('request'),
+    spawn = require('child_process').spawn,
+    gutil = require('gulp-util'),
+    umd = require('gulp-umd'),
+    path = require('path'),
+    karma = require('karma').server,
+    $ = require('gulp-load-plugins')();
+
+// FILES
+var srcFiles = ['src/pact.js', 'src/interaction.js', 'src/http.js', 'src/mockServiceRequests.js', 'src/mockService.js'];
+var specFiles = ['spec/**/*spec.js'];
+var distFiles = ['dist/pact-consumer-js-dsl.js'];
+var karmaConfig = 'spec/karma.conf.js';
+var nodeConfig = 'spec/node.conf.js';
+
+var cleanDirectories = function (directories) {
+    directories.forEach(function (directory) {
+        fs.removeSync(directory);
+        fs.mkdirsSync(directory);
+    });
 };
 
 var waitForServerToStart = function () {
-  var deferred = q.defer();
-  var attempts = 0;
-
-  var checkIfServerHasStarted = function () {
-    attempts += 1;
-    request('http://localhost:1234', function (error) {
-      if (attempts > 100) {
-        deferred.reject(new Error('Timed out waiting for the pact-mock-service to start'));
-      } else if (error) {
-        setTimeout(checkIfServerHasStarted, 100);
-      } else {
-        deferred.resolve();
-      }
-    });
-  };
+    var deferred = q.defer();
+    var attempts = 0;
+
+    var checkIfServerHasStarted = function () {
+        attempts += 1;
+        request('http://localhost:1234', function (error) {
```

Comment:
That is the healthcheck URL. Though it needs the X-Pact-Mock-Service header so that it doesn't throw an "interaction not found" error.